test/configure.js

```diff
@@ -0,0 +1,301 @@
+
+            done();
+        }, data => {
+
+            assert.isArray( data, 'Sugarcoat returns an array.' );
+
+            if ( data.length === 1 ) {
+
+                assert.propertyVal( data[0], 'key', 'sections.title', 'We have sections.title in the error array.' );
+            }
+
+            done();
+        });
+    });
+
+    test( 'Title is set to be required. Title errored out when title and dest were not supplied.', done => {
+
+        var configMissingTitleDest = {
+            sections: [
+                {
+                    files: './test/assert/parseVarCode.css'
+                }
+            ]
+        };
+
+        sugarcoat( configMissingTitleDest )
+        .then( data => {
+
+            done();
+        }, data => {
+
+            assert.isArray( data, 'Sugarcoat returns an array.' );
+
+            for ( var errorObj in data ) {
+
+                var key = data[ errorObj ].key;
+
+                if ( key === 'sections.title' ) {
+
+                    assert.propertyVal( data[ errorObj ], 'key', 'sections.title', 'We have sections.title in the error array.' );
+                    break;
+                }
+            }
+
+            done();
+        });
+    });
+
+    test( 'Title is set to be required. Title errored out when all required options were not supplied.', done => {
+
+        var configMissingTitleFiles = {
+            sections: []
+        };
+
+        sugarcoat( configMissingTitleFiles )
+        .then( data => {
+
+            done();
+        }, data => {
+
+            assert.isArray( data, 'Sugarcoat returns an array.' );
+
+            for ( var errorObj in data ) {
+
+                var key = data[ errorObj ].key;
+
+                if ( key === 'sections.title' ) {
+
+                    assert.propertyVal( data[ errorObj ], 'key', 'sections.title', 'We have sections.title in the error array.' );
+                    break;
+                }
+            }
+
+            done();
+        });
+    });
+
+    test( 'Title is set to be required. Title errored out multiple times when more than one files was not supplied.', done => {
+
+        var configMissingtwoTitles = {
+            settings: {
+                dest: './test/documentation'
+            },
+            sections: [
+                {
+                    files: './test/assert/parseVarCode.css'
+                },
+                {
+                    files: './test/assert/parseVarCode.css'
+                }
+            ]
+        };
+
+        sugarcoat( configMissingtwoTitles )
+        .then( data => {
+
+            done();
+        }, data => {
+
+            assert.isArray( data, 'Sugarcoat returns an array.' );
+
+            for ( var errorObj in data ) {
+
+                var key = data[ errorObj ].key;
+
+                if ( key === 'sections.files' ) {
+
+                    assert.propertyVal( data[ errorObj ], 'key', 'sections.title', 'We have sections.title in the error array.' );
```

Comment:
@ryanfitzer and I just had a conversation about this. We came to the conclusion that we should be erroring out and rejecting on the first error that we encounter. That way, we don't have to worry about adding an errors array to the config, and we don't have to worry about gathering up all of the errors before they are logged out. 

We are also going to use the Error constructor to create an error object that will be passed to npm log and the rejected promise.